### ISSUE TITLE
Using `podman exec --user=root` fails with invalid syntax error

### ISSUE THREAD

**Issue by EmilienM** [containers/libpod#1314]
/kind bug

**Description**

Running a command in a running container fails when specifying a user name but works with UID.

**Steps to reproduce the issue:**

1. Deploy any container with podman.

2. Run: `podman exec my_container ls`. It works.

3. Run: `podman exec --user=0 my_container ls`. It works.

4. Run: `podman exec --user=root my_container ls`. It fails.

**Describe the results you received:**

> exec failed: parsing root as int for uid failed: strconv.Atoi: parsing "root": invalid syntax
> exit status 1

**Describe the results you expected:**

> anaconda-post.log
> bin
> boot
> dev
> etc
> ...

**Output of `podman version`:**

```
Version:       0.8.2.1
Go Version:    go1.10.2
OS/Arch:       linux/amd64
```

**Output of `podman info`:**

```
host:
  Conmon:
    package: podman-0.8.2.1-1.gitf38eb4f.el7.x86_64
    path: /usr/libexec/podman/conmon
    version: 'conmon version 1.12.0-dev, commit: 74bba30adae4fdcacf887d35e4d6331318765028-dirty'
  MemFree: 230641664
  MemTotal: 8201699328
  OCIRuntime:
    package: runc-1.0.0-27.rc5.dev.git4bb1fe4.el7.x86_64
    path: /usr/bin/runc
    version: 'runc version spec: 1.0.0'
  SwapFree: 1610608640
  SwapTotal: 1610608640
  arch: amd64
  cpus: 4
  hostname: undercloud75.localdomain
  kernel: 3.10.0-862.11.6.el7.x86_64
  os: linux
  uptime: 1h 26m 5.42s (Approximately 0.04 days)
insecure registries:
  registries: []
registries:
  registries:
  - docker.io
  - registry.centos.org
  - registry.access.redhat.com
store:
  ContainerStore:
    number: 41
  GraphDriverName: overlay
  GraphOptions:
  - overlay.override_kernel_check=true
  GraphRoot: /var/lib/containers/storage
  GraphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
  ImageStore:
    number: 28
  RunRoot: /var/run/containers/storage
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Running it in local vagrant/virtualbox.

**Comment by mheon** [comment]
@EmilienM Is this specific to rootless, or does it occur when running as root as well?

**Comment by EmilienM** [comment]
@mheon I run in the same situation when executing the command from root.
Note that it works well when using the UUID:

[root@undercloud75 ~]# podman exec --user=0 my_container ls
anaconda-post.log

> bin
> boot
> dev
> etc
> ...

**Comment by mheon** [comment]
It doesn't look like we're doing any translation on the user string, which is probably why numeric IDs are working, but names are not.

I'll need to check against Docker to see where we should be resolving the user - within the container, or on the host. Once we know that should be an easy fix.

**Comment by mheon** [comment]
User looks appear to be happening in the container, so we can use the chrootuser package to perform the lookup.

**Comment by mheon** [comment]
Closed by #1315
